lib/puppet/util/windows/service.rb

```diff
@@ -561,23 +585,45 @@ def query_config(service)
       end
       private :query_config
 
+      # @api private
+      # waits for a windows service to report one of the acceptable_states
+      #
+      # @param [:handle] service handle to the service to wait on
+      # @param [Array<Integer>] acceptable_states array of acceptable states to wait for
+      # @return [bool] 'true' once the service is reporting an acceptable_state,
+      #   'false' if the service never reached one of the acceptable_states
+      def wait_for_state(service, acceptable_states)
+        elapsed_time = 0
+        while elapsed_time <= DEFAULT_TIMEOUT
+          status = query_status(service)
+          state = status[:dwCurrentState]
+          if acceptable_states.include?(state)
+            return true
+          end
+          sleep(1)
+          elapsed_time += 1
+        end
+        return false
+      end
+      private :wait_for_state
+
       # @api private
       # waits for a windows service to report final_state if it
       # is in pending_state
       #
       # @param [:handle] service handle to the service to wait on
-      # @param [Integer] pending_state the state to wait on
+      # @param [Array<Integer>] pending_states array of acceptable states to wait on
       # @param [Integer] final_state the state indicating the transition is finished
       # @return [bool] 'true' once the service is reporting final_state,
-      #   'false' if the service was not in pending_state or finaL_state
-      def wait_for_pending_transition(service, pending_state, final_state)
+      #   'false' if the service never reached final_state or was not in pending_states
+      def wait_on_pending(service, pending_states, final_state)
```

Comment:
Would be nice to spell this out as `wait_on_pending_transition`.

I think we should have only one pending state to strictly enforce the fact that this routine is meant to be used for a _single_ pending transition, even if it still works with multiple pending states. It works for our use-cases.